stl/inc/syncstream

```diff
@@ -0,0 +1,360 @@
+        if (_Wrapped != nullptr) {
+            _Release_mutex_for_instance(_Wrapped);
+            _Wrapped = nullptr;
+        }
+    }
+
+    void _Assign_rv(basic_syncbuf&& _Right) {
+        if (this != _STD addressof(_Right)) {
+            _Tidy();
+            this->_Swap_except_al(_Right);
+        }
+    }
+
+    void _Move_assign(basic_syncbuf&& _Right, _Equal_allocators) noexcept {
+        _Tidy();
+        _Pocma(_Al, _Right._Al);
+        this->_Swap_except_al(_Right);
+    }
+
+    void _Move_assign(basic_syncbuf&& _Right, _Propagate_allocators) noexcept {
+        _Tidy();
+        _Pocma(_Al, _Right._Al);
+        this->_Swap_except_al(_Right);
+    }
+
+    void _Move_assign(basic_syncbuf&& _Right, _No_propagate_allocators) {
+        if (_Al == _Right._Al) {
+            _Move_assign(_Right, _Equal_allocators{});
+        } else {
+            const size_type _Right_buf_size  = _Right._Get_buffer_size();
+            const size_type _Right_data_size = _Right._Get_data_size();
+
+            const pointer _New_ptr = _Al.allocate(_Right_buf_size);
+            _Traits::copy(_New_ptr, pointer_traits<pointer>::pointer_to(*_Right.pbase()), _Right_data_size);
+
+            streambuf_type::setp(_New_ptr, _New_ptr + _Right_data_size, _New_ptr + _Right_buf_size);
+            streambuf_type::_Plocale = _STD move(_Right._Plocale);
+
+            _STD swap(_Mybase::_Emit_on_sync, _Right._Emit_on_sync);
+            _STD swap(_Record_sync, _Right._Record_sync);
+            _STD swap(_Wrapped, _Right._Wrapped);
+
+            _Right._Tidy();
+        }
+    }
+
+    void _Swap_except_al(basic_syncbuf& _Right) noexcept {
+        _Mybase::swap(_Right);
+        _STD swap(_Record_sync, _Right._Record_sync);
+        _STD swap(_Wrapped, _Right._Wrapped);
+    }
+
+    bool _Do_emit() override {
+        return emit();
+    }
+
+    bool _Emit() noexcept {
+        _TRY_BEGIN
+        return emit();
+        _CATCH_ALL
+        return false;
+        _CATCH_END
+    }
+
+    _NODISCARD constexpr size_type _Calculate_growth(
+        const size_type _Oldsize, const size_type _Newsize, const size_type _Maxsize) const {
+        if (_Oldsize > _Maxsize - _Oldsize / 2) {
+            return _Maxsize; // geometric growth would overflow
+        }
+
+        const size_type _Geometric = _Oldsize + _Oldsize / 2;
+
+        if (_Geometric < _Newsize) {
+            return _Newsize; // geometric growth would be insufficient
+        }
+
+        return _Geometric; // geometric growth is sufficient
+    }
+
+    _NODISCARD constexpr size_type _Get_data_size() const noexcept {
+        return static_cast<size_type>(streambuf_type::pptr() - streambuf_type::pbase());
+    }
+
+    _NODISCARD constexpr size_type _Get_buffer_size() const noexcept {
+        return static_cast<size_type>(streambuf_type::epptr() - streambuf_type::pbase());
+    }
+
+    streambuf_type* _Wrapped{nullptr};
+    allocator_type _Al{};
+    bool _Record_sync{false};

```

Comment:
`_Record_sync` should be moved up to `_Basic_syncbuf_impl` to avoid padding. Also `_Sync_recorded` might be a better name.